src/ucm/mmap/install.c

```diff
@@ -31,10 +31,27 @@
 #include <unistd.h>
 #include <pthread.h>
 
+#if defined(__riscv)
+ #include <ucs/arch/cpu.h>
+#endif
 
 #define UCM_HOOK_STR \
     ((ucm_mmap_hook_mode() == UCM_MMAP_HOOK_RELOC) ?  "reloc" : "bistro")
 
+#if defined(__riscv)
+#define UCM_FIRE_EVENT(_event, _mask, _data, _call)                            \
+     do {                                                                      \
+         int exp_events = (_event) & (_mask);                                  \
+         (_data)->fired_events = 0;                                            \
+         ucs_rv64_isb();                                                       \
```

Comment:
Hmm, why ISB is needed here ? I don't see any obvious reason.
If there is good reason probably it is required on all architectures.

Reply:
More I read it  and more I don't understand it. The ISB if followed by system call that switches to privilege mode. In theory it should  include ISB and memory barrier.  Out curiosity, have you tried to replace it with memory write barrier ?